packages/maskbook/src/plugins/NFT/UI/NFTInPost.tsx

```diff
@@ -0,0 +1,42 @@
+import { useEffect, useState } from 'react'
+import MaskbookPluginWrapper from '../../MaskbookPluginWrapper'
+import type { NFTInPostProps, TokenDetails } from '../types'
+import { parseNftUrl } from '../utils'
+import NFTCardUI from './NFTCardUI'
+
+export default function NFTInPost(props: NFTInPostProps) {
+    const [tokenDetails, setTokenDetails] = useState<TokenDetails>({})
+
+    useEffect(function () {
```

Comment:
You can use `useAsync` from `react-use`. It will be easier to write async functions